Activities/FTP/UiPath.FTP.Activities.Design/ActivitySynonymApiRegistration.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using UiPath.Studio.Activities.Api;
+using UiPath.FTP.Activities.Design.Properties;
+
+namespace UiPath.FTP.Activities.Design
+{
+    /// <summary>
+    /// Class for synonyms registration
+    /// </summary>
+    internal class ActivitySynonymApiRegistration : ApiRegistrationBase
+    {
+        /// <summary>
+        /// Checks whether the registration of synonyms can be performed
+        /// </summary>
+        /// <param name="api">The api</param>
+        /// <returns>True or false, meaning whether the registration can be performed or not</returns>
+        public override bool CanPerformRegistration(IWorkflowDesignApi api)
+        {
+            return api != null && api.HasFeature(DesignFeatureKeys.ActivitySynonyms);
+        }
+
+        /// <summary>
+        /// Register synonyms for the required activities
+        /// </summary>
+        /// <param name="api">The api</param>
+        protected override void PerformRegistration(IWorkflowDesignApi api)
```

Comment:
Please add `[MethodImpl(MethodImplOptions.NoInlining)]` to this method.

See how other activity packs do this (e.g. Mail in the main Activities repo).